### PR TITLE
[generator2] split grammar and CA grammar

### PR DESCRIPTION
**Pull request by oehme**

### CODE REVIEW

File: plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AbstractAntlrGrammarGenerator.xtend [hunk 1]
```diff
@@ -45,30 +45,57 @@ abstract class AbstractAntlrGrammarGenerator {
...
-	protected def compile(Grammar it, AntlrOptions options) '''
+	protected def compileParser(Grammar it, AntlrOptions options) '''
 		�codeConfig.fileHeader�
-		grammar �grammarNaming.getGrammarClass(it).simpleName�;
-		�compileOptions(options)�
-		�compileTokens(options)�
-		�compileLexerHeader(options)�
+		�IF !combinedGrammar�parser �ENDIF�grammar �grammarNaming.getParserGrammar(it).simpleName�;
+		�compileParserOptions(options)�
+		�IF isCombinedGrammar�
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AbstractAntlrGrammarGenerator.xtend]: Two subsequent checks for `isCombinedGrammar` - is this intentional? Merge both cases?

Comment (oehme) [inline plugins/org.eclipse.xtext.xtext.generator/src/org/eclipse/xtext/xtext/generator/parser/antlr/AbstractAntlrGrammarGenerator.xtend]: This was unintentional

**Comment by oehme** [comment]
We need to have another look at the grammar generators. The CA generator is very redundant at the moment, because the methods from the superclass didn't really fit. But I didn't want to write it *and* restructure the production parser at the same time.

**Comment by oehme** [comment]
Jenkins test

**Pull request merged**